subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java

```diff
@@ -102,22 +102,26 @@ public UnitOfWork getWork() {
     }
 
     private static IncrementalInputProperties createIncrementalInputProperties(UnitOfWork work) {
-        if (!work.isRequiresInputChanges()) {
-            return IncrementalInputProperties.NONE;
+        UnitOfWork.Incrementality incrementality = work.getIncrementality();
```

Comment:
I think `UnitOfWork.isIncremental()` should be more about `getInputChangeTrackingStrategy()` and return something like:

```java
enum InputChangeTrackingStrategy {
    INCREMENTAL_PARAMETERS,
    NONE,
    @Deprecated
    ALL_PARAMETERS
}
```

Reply:
You could also add an `isIncremental()` method.

Reply:
Renamed and added a method.